flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/runtime/metadata/SavepointMetadata.java

```diff
@@ -20,32 +20,38 @@
 
 import org.apache.flink.annotation.Internal;
 import org.apache.flink.runtime.checkpoint.MasterState;
-import org.apache.flink.runtime.checkpoint.OperatorState;
 
-import java.io.IOException;
 import java.io.Serializable;
 import java.util.Collection;
 
 /**
  * Returns metadata about a savepoint.
  */
 @Internal
-public interface SavepointMetadata extends Serializable {
+public class SavepointMetadata implements Serializable {
```

Comment:
I don't think we need this base class anymore (at least from the current state of the code). It's only subclass is the `ModifiableSavepointMetadata`.

Moreover, the contents of this base class doesn't seem to justify as a proper savepoint metadata anyways, since it only maintains the master states.

Therefore, I would suggest to simplify the hierarchy and just have a single `SavepointMetadata` class for now.